From 4adf7d72219a5c79facbd6a37d9c34eb828ed83c Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:51:06 -0400
Subject: [fs] nfsd: minor encode_read cleanup

Message-id: <1403560314-32525-125-git-send-email-bfields@redhat.com>
Patchwork-id: 84602
O-Subject: [Patch RHEL7 124/172] nfsd4: minor encode_read cleanup
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 6ff9897d2bcf4036dfd139caeddd6f0a51c9ca06
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4xdr.c b/fs/nfsd/nfs4xdr.c
index 8b3d24d..8ce6c8d 100644
--- a/fs/nfsd/nfs4xdr.c
+++ b/fs/nfsd/nfs4xdr.c
@@ -3076,18 +3076,20 @@ nfsd4_encode_read(struct nfsd4_compoundres *resp, __be32 nfserr,
 
  len = maxcount;
  v = 0;
- while (len > 0) {
+ while (len) {
+  int thislen;
+
   page = *(resp->rqstp->rq_next_page);
   if (!page) { /* ran out of pages */
    maxcount -= len;
    break;
   }
+  thislen = min_t(long, len, PAGE_SIZE);
   resp->rqstp->rq_vec[v].iov_base = page_address(page);
-  resp->rqstp->rq_vec[v].iov_len =
-   len < PAGE_SIZE ? len : PAGE_SIZE;
+  resp->rqstp->rq_vec[v].iov_len = thislen;
   resp->rqstp->rq_next_page++;
   v++;
-  len -= PAGE_SIZE;
+  len -= thislen;
  }
  read->rd_vlen = v;
 
-- 
1.7.1